src/tarski/syntax/terms.py

```diff
@@ -187,6 +198,167 @@ def is_syntactically_equal(self, other):
         # Else we just need to recursively check if all subterms are syntactically equal
         return all(x.is_syntactically_equal(y) for x, y in zip(self.subterms, other.subterms))
 
+
+class AggregateCompoundTerm(Term):
```

Comment:
What is exactly an Aggregate Compound Term? I read the doc below but not sure what it means!

Reply:
Improved comments on commit  2880b55f61d23a79e12bb965451a4eb22b6b4abd 

```
Combinations of a functional symbol, a set of bound variables and a sequence of terms.
Examples of such terms include summations, products and other operations over sequences.
```

Here I was imitating ```yices``` - where yices has a function for each term type, we have a class

```
term_t yices_sum(uint32_t n, const term_t t[])
Constructs the sum (+ t[0] … t[n-1]).

Parameters

n is the size of array t
t must be an array of n arithmetic terms
This generalizes function yices_add to n arguments. The array may be empty (i.e., n may be zero), in which case the function returns 0.

term_t yices_product(uint32_t n, const term_t t[])
Constructs the product (* t[0] … t[n-1]).

Parameters

n is the size of array t
t must be an array of n arithmetic terms
This generalizes function yices_mul to n arguments. If n is zero, the function returns 1.
```